packages/react/src/transactions/adapters/PublicationCallGateway.ts

```diff
@@ -226,6 +290,30 @@ export class PublicationCallGateway implements ICreatePostCallGateway {
     );
   }
 
+  private async createCommentCall<T extends CreateCommentRequest>(
```

Comment:
I know in the original codebase I modelled all these as methods of the same class... I think I violated Open/Closed Principle in doing so and the code seems complicated. Is it thinkable to create an abstract class that a Post Gateway and a Comment Gateway extends by filling the gaps? Will it be easier to manage? Also "create mirror" beyond creating a Lens Publication does have little overlap with the settings of Post and Comment.

Thoughts?

Reply:
Seems there were no use cases for an abstract class as post/comments are both following separate code paths. 
They share some common utils (which are already separate functions) and upload handler logic (which for now is very simple, once #58 is merged we would need to create a new `UploadHandler` class that handles catching and rehrowing upload errors) so I refactored them as 2 separate classes.

Reply:
Composition over inheritance feels a better choice, I totally agree.